packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,290 @@
+    let callCount = 0;
+
+    const wrapperFunc = () => {
+        callCount++;
+        return func(callCount);
+    };
+
+    return {
+        funcToRetry: wrapperFunc,
+        getAttempts: () => {
+            return callCount;
+        },
+    };
+};
+
+describe('chai-retry-plugin', () => {
+    it('should retry a function that eventually succeeds', async () => {
+        const { funcToRetry, getAttempts } = getFunctionToRetry((attempts: number) => {
+            if (attempts < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        });
+
+        await expect(funcToRetry).to.retry().to.equal('Success');
+        expect(getAttempts()).to.equal(3);
+    });
+
+    it('should retry a function that always fails and throw an error', async () => {
+        const { funcToRetry, getAttempts } = getFunctionToRetry(() => {
+            throw new Error('Failed');
+        });
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect((error as Error).message).to.includes('Limit of 3 retries');
+            expect(getAttempts()).to.equal(3);
+        }
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await sleep(250);
+                return 'Success';
+            };
+
+            try {
+                await expect(funcToRetry).retry({ timeout: 700 }).to.equal('Failure');
+                throw new Error('This should not be called');
+            } catch (error: unknown) {
+                expect((error as Error).message).includes('Timed out after 700ms');
+            }
+        });
+
+        it('throw an error when retries limit exceeded', async () => {
+            const { funcToRetry } = getFunctionToRetry((attempts) => attempts);
+
+            try {
+                await expect(funcToRetry).retry({ retries: 10 }).to.be.above(100);
+                throw new Error('This should not be called');
+            } catch (error: unknown) {
+                expect((error as Error).message).to.includes('Limit of 10 retries exceeded!');
+            }
+        });
+
+        it('should apply delay correctly', async () => {
+            let attempts = 0;
+            let end = 0;
+            const funcToRetry = () => {
+                attempts++;
+
+                if (attempts === 5) {
+                    end = Date.now();
+                }
+
+                return attempts;
+            };
+
+            const start = Date.now();
+            await expect(funcToRetry).retry({ delay: 100 }).to.equal(5);
+
+            const elapsed = end - start;
+            const lowerBound = 400;
+            const upperBound = 450;
+
+            expect(elapsed).to.be.within(
```

Comment:
This test is more complex than it should be. Just make function immediately throw,  set `timeout` to 1000, `delay` to 100 and verify that function has been called 9 or 10 times

Reply:
@yurii-ve fixed